internal/pkg/service/objectstorage/cloud/azure/storage.go

```diff
@@ -276,139 +184,5 @@ func (s AzureObjectStorageAdapter) DeleteBucket(ctx context.Context, bucket *v1a
-								Type:    to.Ptr(armstorage.RuleTypeLifecycle),
-								Definition: &armstorage.ManagementPolicyDefinition{
-									Actions: &armstorage.ManagementPolicyAction{
-										BaseBlob: &armstorage.ManagementPolicyBaseBlob{
-											Delete: &armstorage.DateAfterModification{
-												DaysAfterModificationGreaterThan: to.Ptr[float32](float32(bucket.Spec.ExpirationPolicy.Days)),
-											},
-										},
-									},
-									Filters: &armstorage.ManagementPolicyFilter{
-										BlobTypes: []*string{
-											to.Ptr("blockBlob"),
-										},
-									},
-								},
-							},
-						},
-					},
-				},
-			},
-			nil,
-		)
-		if err != nil {
-			s.logger.Error(err, fmt.Sprintf("Error creating/updating Policy Rule for Storage Account %s", storageAccountName))
-		}
-		return err
-	}
-
-	// No Lifecycle Policy defines in the bucket CR, we delete it in the Storage Account
-	_, err := s.managementPoliciesClient.Delete(
-		ctx,
-		s.cluster.GetResourceGroup(),
-		storageAccountName,
-		LifecycleRuleName,
-		nil,
-	)
-	if err != nil {
-		var respErr *azcore.ResponseError
-		if errors.As(err, &respErr) {
-			// If the Lifecycle policy does not exists, it's not an error
-			if respErr.StatusCode == http.StatusNotFound {
-				return nil
-			}
-		}
-	}
-	return err
-}
-
-func sanitizeTagKey(tagName string) string {
-	return strings.ReplaceAll(tagName, "-", "_")
-}
-
-// setTags set cluster additionalTags and bucket tags into Storage Container Metadata
-func (s AzureObjectStorageAdapter) setTags(ctx context.Context, bucket *v1alpha1.Bucket) error {
-	storageAccountName := s.getStorageAccountName(bucket.Spec.Name)
-	tags := map[string]*string{}
-	for _, t := range bucket.Spec.Tags {
-		// We use this to avoid pointer issues in range loops.
-		tag := t
-		if tag.Key != "" && tag.Value != "" {
-			tags[sanitizeTagKey(tag.Key)] = &tag.Value
-		}
-	}
-	for k, v := range s.cluster.GetTags() {
-		// We use this to avoid pointer issues in range loops.
-		key := k
-		value := v
-		if key != "" && value != "" {
-			tags[sanitizeTagKey(key)] = &value
-		}
-	}
-
-	// Updating Storage Container Metadata with tags (cluster additionalTags + Bucket tags)
-	_, err := s.blobContainerClient.Update(
-		ctx,
-		s.cluster.GetResourceGroup(),
-		storageAccountName,
-		bucket.Spec.Name,
-		armstorage.BlobContainer{
-			ContainerProperties: &armstorage.ContainerProperties{
-				Metadata: tags,
-			},
-		},
-		nil,
-	)
-	if err != nil {
-		s.logger.Error(err, fmt.Sprintf("Error updating Storage Container %s Metadata", bucket.Spec.Name))
-	}
-	return err
-}
-
-// getStorageAccountName returns the storage account name for the given bucket name.
-// It sanitizes the bucket name and checks if it already exists in the list of storage account names.
-// If it exists, it returns the sanitized name. Otherwise, it adds the sanitized name to the list and returns it.
-func (s *AzureObjectStorageAdapter) getStorageAccountName(bucketName string) string {
```

Comment:
Are we no longer checking for existence of the bucket name ? why ?

Reply:
No we are through the existsContainer and existStorageAccount functions